### DISCUSSION TITLE
c interface api ?

### DISCUSSION THREAD

**Discussion by calvin2021y** [slint-ui/slint#6275] [Ideas]
Is there a plan to add c api interface support ?

with C api, it will be able to add php, lua, py and a lot other languages support.

**Comment by ogoffart** [comment]
Sort of. 
For the purpose of the C++ bindings, we generate a C++ interface in https://github.com/slint-ui/slint/blob/master/api/cpp/cbindgen.rs
But it wouldn't be too hard to make it export a C interface instead.
The question is weather it'd be best to use a C interface or if these language can have an interface with rust.  For example, we do support python, but using pyo3 rather than C headers.
